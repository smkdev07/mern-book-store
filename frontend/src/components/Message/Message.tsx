import React from 'react';

import { Alert } from 'react-bootstrap';

type AlertVariants =
  | 'primary'
  | 'secondary'
  | 'success'
  | 'danger'
  | 'warning'
  | 'info'
  | 'light'
  | 'dark';

export interface MessageProps {
  variant?: AlertVariants;
  children: React.ReactNode;
}

const Message: React.FC<MessageProps> = ({ variant = 'info', children }) => (
  <Alert variant={variant}>{children}</Alert>
);

// Message.defaultProps = {
//   variant: 'info',
// };

export default Message;
