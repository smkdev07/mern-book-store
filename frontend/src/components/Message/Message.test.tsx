import React from 'react';
import { shallow } from 'enzyme';

import Message, { MessageProps } from './Message';

const setup = (props: MessageProps) => {
  return shallow(<Message {...props} />);
};

describe('<Message />', () => {
  it('renders the <Message /> component with default variant', () => {
    const childElement = <p>Test Message</p>;
    const wrapper = setup({ children: childElement });
    expect(wrapper).toMatchSnapshot();
  });

  it('renders the <Message /> component with variant from props', () => {
    const variant = 'primary';
    const childElement = <p>Test Message</p>;
    const wrapper = setup({ variant, children: childElement });
    expect(wrapper).toMatchSnapshot();
  });
});
