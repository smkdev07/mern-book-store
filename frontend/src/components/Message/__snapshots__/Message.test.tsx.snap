// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Message /> renders the <Message /> component with default variant 1`] = `ShallowWrapper {}`;

exports[`<Message /> renders the <Message /> component with variant from props 1`] = `ShallowWrapper {}`;
